automation/src/automation/bootstraps/pursuedpybear.py

```diff
@@ -0,0 +1,83 @@
+import sys
+
+import tomli_w
+
+if sys.version_info >= (3, 11):
+    import tomllib
+else:
+    import tomli as tomllib
+
+from automation.bootstraps import BRIEFCASE_EXIT_SUCCESS_SIGNAL, EXIT_SUCCESS_NOTIFY
+from briefcase.bootstraps import PursuedPyBearGuiBootstrap
+
+
+class PursuedPyBearAutomationBootstrap(PursuedPyBearGuiBootstrap):
+    def app_source(self):
+        return f"""\
+import importlib.metadata
+import os
+import sys
+
+import ppb
+
+
+class {{{{ cookiecutter.class_name }}}}(ppb.Scene):
+    def __init__(self, **props):
+        super().__init__(**props)
+        self.updates: int = 0
+
+        self.add(
+            ppb.Sprite(
+                image=ppb.Image("{{{{ cookiecutter.module_name }}}}/resources/{{{{ cookiecutter.app_name }}}}.png"),
+            )
+        )
+
+    def on_update(self, event, signal):
+        self.updates += 1
+        # quit after 2 seconds since on_update is run 60 times/second
+        if self.updates > 120:
+            print("{EXIT_SUCCESS_NOTIFY}")
+            print("{BRIEFCASE_EXIT_SUCCESS_SIGNAL}")
+            signal(ppb.events.Quit())
+
+
+def main():
+    # Linux desktop environments use an app's .desktop file to integrate the app
+    # in to their application menus. The .desktop file of this app will include
+    # the StartupWMClass key, set to app's formal name. This helps associate the
+    # app's windows to its menu item.
+    #
+    # For association to work, any windows of the app must have WMCLASS property
+    # set to match the value set in app's desktop file. For PPB, this is set
+    # using the SDL_VIDEO_X11_WMCLASS environment variable.
+
+    # Find the name of the module that was used to start the app
+    app_module = sys.modules["__main__"].__package__
+    # Retrieve the app's metadata
+    metadata = importlib.metadata.metadata(app_module)
+
+    os.environ["SDL_VIDEO_X11_WMCLASS"] = metadata["Formal-Name"]
+
+    ppb.run(
+        starting_scene={{{{ cookiecutter.class_name }}}},
+        title=metadata["Formal-Name"],
+    )
+"""
+
+    # The constraint of pysdl2-dll==2.0.22 is required for ppb==1.1.0;
+    # the libraries in later versions of pysdl2-dll are not compatible.
+
+    def pyproject_table_linux_flatpak(self):
+        table = tomllib.loads(super().pyproject_table_linux_flatpak())
+        table.setdefault("requires", []).append("pysdl2-dll==2.0.22")
```

Comment:
Why is this only part of the automation, and not the core ppb bootstrap?

Reply:
Mostly just because I wasn't sure if we wanted to officially endorse this for new projects....but I'm probably only really being conservative because I don't know anything about the ppb project.

That said, it seems fairly harmless to add this to the actual bootstrap.

Reply:
Probably worth tagging @pathunstrom to see if she has any thoughts on the matter.

/me turns on the bat signal...

Reply:
Via Discord: @pathunstrom has indicated [the patch is on the right track](https://discord.com/channels/836455665257021440/836455665257021443/1181375691401461761); we'll likely need to revisit it when PPB can address the problem at their end, but for now, it works.